src/service.ts

```diff
@@ -400,24 +410,46 @@ export class GrpcService extends Service {
    * @param {object} reqOpts - The request options.
    * @param {function=} callback - The callback function.
    */
-  // @ts-ignore
-  request(protoOpts, reqOpts, callback) {
+  request(reqOpts: DecorateRequestOptions): Promise<r.Response>;
+  request(reqOpts: DecorateRequestOptions, callback: BodyResponseCallback):
+      void;
+  request(reqOpts: DecorateRequestOptions, callback?: BodyResponseCallback):
+      void|Promise<r.Response>;
+  request(
+      protoOpts: ProtoOpts, reqOpts: DecorateRequestOptions,
+      callback: ServiceRequestCallback): Abortable|void;
+  request(
+      pOpts: ProtoOpts|DecorateRequestOptions,
+      rOpts?: DecorateRequestOptions|BodyResponseCallback,
+      callback?: ServiceRequestCallback): Abortable|void|Promise<r.Response> {
+    /**
+     * The function signature above is a little funky.  This is due to the way
+     * method overloading in TypeScript operates.  Since this class extends
+     * ServiceObject, the signatures for `request` need to have
```

Comment:
Did you mean `Service` instead of `ServiceObject`?

Reply:
I did. Thank you :)